### ISSUE TITLE
Add features and UI

### ISSUE THREAD

**Issue by MaddulaSamanvitha** [nupurmadaan04/AI-agriculture-yield-production#1]
1. Feature & Model Enhancements:

Also, try including recent patterns in crop growth instead of very old data from 2010-2015.

Add More Features: Include weather data (rainfall, temperature), soil quality, or fertilizer usage.

State-wise Modeling: Create state-specific models or add state-wise interaction features.

2. Try Other ML Models: Try with various ML algorithms and also try applying hyperparameter tuning

1) Gradient Boosting (XGBoost, LightGBM)
2) CatBoost (works well with categorical data)
3) Support Vector Regressor (for smaller subsets)

3. User Interface / Visualization
🌐 Build an Interactive Dashboard:
Use Streamlit to build a web UI where:

Users select state/year/area/production → get predicted yield

Graphs visualize yield trends and comparisons.

**Comment by nupurmadaan04** [comment]
@MaddulaSamanvitha That's glad to hear that you are enthusiasm towards this project , but i believe a single contributor can't solve all these issues let's split these , and would you like to work on any of these issues?

**Comment by MaddulaSamanvitha** [comment]
My domain of knowledge is AI/ML any task related to it I can handle

**Comment by nupurmadaan04** [comment]
@MaddulaSamanvitha Hey , right now we have Issue #15 unassigned u can check if u can work on this. If not then no need to worry am personally analysing old data and one the basis of analysis I'll create more issues . Till then u can star repository for updates.

@GxAditya Hey Aditya , UI design work is assigned rn to someone else , u can check the files and code if it seems that there's need of anything else then u can create issue. And if the contributor that has been assigned ui work will not be able to do work properly then I can assign you.

**Comment by MaddulaSamanvitha** [comment]
Hey @nupurmadaan04 I have checked issue 15 it's actually a new concept to me but I am ready to explore and work on it if it's OK for you

**Comment by nupurmadaan04** [comment]
@MaddulaSamanvitha I got you , it's upto you dear if u want to explore or not , it's unassigned for now . But we r working on project and on the basis of analysis we'll create issues related to AI/ML and DS

**Comment by MaddulaSamanvitha** [comment]
Yeah I can try working on it but is it possible to assign a mentor or a contributor for a collaborative approach from your end?
Like make contributors interactive with each other and all especially or with a mentor also is fine for me

**Comment by nupurmadaan04** [comment]
@MaddulaSamanvitha Yes for sure , I'll share the mentor's LinkedIn profile in DM on discord, just ping a message there.

**Comment by MaddulaSamanvitha** [comment]
Yes sure thank you for the consideration

**Comment by nupurmadaan04** [comment]
@MaddulaSamanvitha just comment on issue #15 so I can assign you that issue , one more contributor is willing to do so you both can collaboratively work on the issue.
